### PR TITLE
config: migrate deprecated recordings options, add localRecording sec…

### PR DESCRIPTION
**Pull request by debendraoli**
…tion and disableProfile

### CODE REVIEW

File: web/rootfs/defaults/settings-config.js [hunk 1]
```diff
@@ -152,22 +155,22 @@ config.dropbox.redirectURI = '{{ .Env.DROPBOX_REDIRECT_URI }}';
...
 // (e.g. meeting participants), based on the actual implementation
 // on the backend.
-config.fileRecordingsServiceSharingEnabled = true;
+config.recordingService.sharingEnabled = true;
 {{ end -}}
 {{ end -}}
 
 
+// Local recording configuration.
+{{ if $DISABLE_LOCAL_RECORDING -}}
```
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:168]: Don't add a condition if the setting is boolean, set it always.

File: web/rootfs/defaults/settings-config.js [hunk 2]
```diff
@@ -307,6 +310,11 @@ config.requireDisplayName = {{ $ENABLE_REQUIRE_DISPLAY_NAME }};
 config.chromeExtensionBanner = {{ .Env.CHROME_EXTENSION_BANNER_JSON }};
 {{ end -}}
 
+// Disables profile and the edit of all fields from the profile settings (display name and email)
+{{ if $DISABLE_PROFILE -}}
```
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:314]: Ditto

File: web/rootfs/defaults/settings-config.js [hunk 3]
```diff
@@ -132,10 +133,12 @@ config.etherpad_base = '{{ $PUBLIC_URL }}/etherpad/p/';
 
 {{ if $ENABLE_RECORDING -}}
 
+if (!config.hasOwnProperty('recordingService')) config.recordingService = {};
```
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:136]: Can you move this one line below please? So all options are grouped.

**Review by saghul — CHANGES_REQUESTED** [review]

File: web/rootfs/defaults/settings-config.js [hunk 4]
```diff
@@ -152,22 +155,20 @@ config.dropbox.redirectURI = '{{ .Env.DROPBOX_REDIRECT_URI }}';
...
-// depends on jibri configuration)
-config.fileRecordingsServiceEnabled = true;
-{{ end -}}
-{{ if $ENABLE_FILE_RECORDING_SERVICE_SHARING -}}
+{{ if $ENABLE_FILE_RECORDING_SHARING -}}
 // Whether to show the possibility to share file recording with other people
 // (e.g. meeting participants), based on the actual implementation
 // on the backend.
-config.fileRecordingsServiceSharingEnabled = true;
+config.recordingService.sharingEnabled = true;
```
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:162]: Ditto, set this to the value of the variable, unconditionally.

File: web/rootfs/defaults/settings-config.js [hunk 5]
```diff
@@ -152,22 +155,20 @@ config.dropbox.redirectURI = '{{ .Env.DROPBOX_REDIRECT_URI }}';
...
 {{ end -}}
 {{ end -}}
 
 
+// Local recording configuration.
+if (!config.hasOwnProperty('localRecording')) config.localRecording = {};
+config.localRecording.disable = $DISABLE_LOCAL_RECORDING;
```
Comment (saghul) [inline web/rootfs/defaults/settings-config.js:169]: Can you add variables for the rest of the settings, while you're here?

Comment (debendraoli) [reply web/rootfs/defaults/settings-config.js:169]: These setting seems disabled by default.

What would be the environment variable names ?

```
DISABLE_LOCAL_RECORDING_NOTIFY_ALL_PARTICIPANT
DISABLE_LOCAL_RECORDING_SELF_RECORDING
```
or
```
ENABLE_LOCAL_RECORDING_NOTIFY_ALL_PARTICIPANT
ENABLE_LOCAL_RECORDING_SELF_RECORDING
```

**Comment by debendraoli** [comment]
Is there anythings I can do to get merged?

**Review by saghul — APPROVED** [review]

**Pull request merged**